### DISCUSSION TITLE
Is there a limit on number of simultaneous smtp connections?

### DISCUSSION THREAD

**Discussion by chrism417** [postalserver/postal#2072] [Help with using Postal]
We were load testing postal and in the last 12 hours we've sent roughly 126k emails through it. In production this will be at least double that amount, if not more. We developed a backlog in O365 relaying email through postal with the error: 

Office 365 closed the connection with the destination messaging.

I had an HPA setup to scale workers based on CPU utilization, but they didn't seem to be doing much. After seeing the above message, I knew it had to be smtp. smtp resources were barely being touched, but I went from 2 smtp pods to 10 and the backlog (4300 emails) was gone in 15 minutes and the alert cleared in O365.

**Comment by willpower232** [comment]
It is interesting that the HPA didn't react, I'm not sure how useful the original config settings are towards actually processing faster

https://github.com/postalserver/postal/blob/9f0697f194209f5fae5e451ba8fb888413fe37fa/config/postal.defaults.yml#L61-L63

**Comment by chrism417** [comment]
well, the workers were scaled up when we had a ton of emails coming through, however it was a steady flow and the number backlogged continued to rise. It was until I scaled up the number of smtp pods we chewed through the backlog.

Regarding:
` workers: 
   quantity: 1 
   threads: 4 `

Does this have any effect on smtp or only the worker?

**Reply by willpower232** [reply]
Oh I fully misunderstood what you're doing, you're trying to get lots of incoming mail into Postal. Yeah there is definitely some awkwardness around there, you might be interested in https://github.com/postalserver/postal/pull/751 but I'm not really sure why you would relay from 365 via Postal to the internet.

**Reply by chrism417** [reply]
Basically we're going to use postal as a "front end" to all our internal mail servers. The way this one particular service works is it's a journal address, so all emails sent to a domain are then journaled. We want front-end metrics on how many incoming emails are going to this address, so the emails are being directed to postal, then to their original destination.

**Reply by willpower232** [reply]
I guess that works, Postal wasn't really designed to store messages long term but you might be able to get some vague numbers from it. I'd definitely keep more than a few SMTP pods around because you really need that high availability.
